iot-hub/Quickstarts/read-d2c-messages/README.md

```diff
@@ -0,0 +1,58 @@
+# Read device-to-cloud messages
+
+This sample demonstrates how to use the Azure Event Hubs client library for .NET to read messages sent from a device by using the built-in Event Hub that exists by default for every IoT Hub instance. 
+
+## Prerequisites
+
+The .NET Core SDK 3.0.0 or greater is recommended.  You can download the .NET Core SDK for multiple platforms from [.NET](https://www.microsoft.com/net/download/all).  You can verify the current version of C# on your development machine using 'dotnet --version'.
+
+This sample can also be compiled using the NET Core SDK 2.1 SDK if the language version of project is changed to `preview`.
+
+## Obtain the Event Hubs-compatible connection string
+
+You can get the Event Hubs-compatible connection string to your IotHub instance via the Azure portal or
+by using the Azure CLI.
+
+If using the Azure portal, see [Built in endpoints for IotHub](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-read-builtin#read-from-the-built-in-endpoint) to get the Event Hubs-compatible connection string and assign it to the variable `connectionString` in the sample. You can skip the Azure CLI instructions in the sample after this.
+
+If using the Azure CLI, you will need to run the below before running this sample to get the details required to form the Event Hubs compatible connection string:
+
+```bash
+az iot hub show --query properties.eventHubEndpoints.events.endpoint --name {your IoT Hub name}
+az iot hub show --query properties.eventHubEndpoints.events.path --name {your IoT Hub name}
+az iot hub policy show --name service --query primaryKey --hub-name {your IoT Hub name}
+```
+
+If you can do neither of the above and need to programmatically get this information, the sample [How to request the IoT Hub built-in Event Hubs-compatible endpoint connection string](https://github.com/Azure/azure-sdk-for-net/blob/master/samples/iothub-connect-to-eventhubs/README.md) demonstrates how to do so.
+
+## WebSocket and proxy support
+
+If you would like to use WebSockts, with our without a proxy, you will need to create a set of options for the `EventHubConsumerclient` to configure its behavior.  Proxy support is offered via the `IWebProxy` interface, which includes the built-in [`WebProxy`](https://docs.microsoft.com/en-us/dotnet/api/system.net.webproxy?view=netcore-3.1) class.  Any proxy must be explicitly passed; the client does not assume that any proxy set via the ambient environment or system-wide is desired.
+
+The options may be created as follows:
+
+```csharp
+var options = new EventHubConsumerClientOptions();
+
+// This line sets the transport to use WebSockets.
+options.ConnectionOptions.TransportType = EventHubsTransportType.AmqpWebSockets;
+
+// The following lines configure the options for proxy use.
+IWebProxy proxy = { SUPPLY YOUR PROXY };
```

Comment:
Wouldn't this generally be an instance of [WebProxy](https://docs.microsoft.com/en-us/dotnet/api/system.net.webproxy?view=netcore-3.1)? If so, can we be a little more specific than {SUPPLY YOUR PROXY}?

Also, in addition to the comments in the readme, the samples on the other platforms include some example code for WebSockets and proxy (commented out where necessary). Is it possible to do this for csharp? For example: https://github.com/Azure-Samples/azure-iot-samples-python/pull/24/files or https://github.com/Azure-Samples/azure-iot-samples-java/pull/15/files

Reply:
Fair enough.   Updated.